AK/PrintfImplementation.h

```diff
@@ -283,166 +285,227 @@ ALWAYS_INLINE int print_signed_number(PutChFunc putch, char*& bufptr, int number
     return print_number(putch, bufptr, number, left_pad, zero_pad, field_width);
 }
 
-template<typename PutChFunc>
-ALWAYS_INLINE int printf_internal(PutChFunc putch, char* buffer, const char*& fmt, va_list ap)
+struct State {
```

Comment:
It's not really State, it's more like "modifiers" or "display options". But of course those display options are accumulated as we scan through the format string, so there's a point in calling it State too.

Perhaps you can think of a name that represents both?

Reply:
`OptionState`?
`AccumulatedDisplayOptions`?

I dunno, I'm not great at naming things

Reply:
Actually `ModifierState` sounds pretty good to me.

Reply:
`ModifierState` it is, then.